yt/data_objects/construction_data_containers.py

```diff
@@ -969,13 +969,31 @@ class YTSmoothedCoveringGrid(YTCoveringGrid):
     filename = None
     _min_level = None
     @wraps(YTCoveringGrid.__init__)
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, interp='linear', nbuf=None, **kwargs):
         ds = kwargs['ds']
         self._base_dx = ((ds.domain_right_edge - ds.domain_left_edge) /
                          ds.domain_dimensions.astype("float64"))
         self.global_endindex = None
         YTCoveringGrid.__init__(self, *args, **kwargs)
         self._final_start_index = self.global_startindex
+        # Assign order (covergence) based on interpolation method
+        if interp == 'linear': 
+            order=2
+        elif interp == 'cubic': 
+            order=4
+        else:
+            raise NotImplementedError
+        # Assess buffer region surrounding grid
+        if nbuf == None:
+            if order > 0: 
+                nbuf = order//2
+            elif order == -1:
+                nbuf = 3
+        else:
+            assert nbuf >= 1
```

Comment:
reasons:
- `None` is a singleton, using `==` is bad practice
- don't use `assert` in business-logic code (because assert statements can be ignored with the `-O` flag of the Python executable, no matter how unlikely the actual usage)

Other remarks:
- I'm not sure how `order` could end up being negative, seems impossible in this implementation
- shouldn't the `nbuf` variable depend on the dimensionality of the data somehow ?

```suggestion
        if nbuf is None:
            if order > 0: 
                nbuf = order//2
            elif order == -1:
                nbuf = 3
        elif nbuf < 1:
           raise ValueError
```